src/input/datetime.rs

```diff
@@ -264,8 +265,17 @@ pub fn bytes_as_date<'a>(input: &'a impl Input<'a>, bytes: &[u8]) -> ValResult<'
     }
 }
 
-pub fn bytes_as_time<'a>(input: &'a impl Input<'a>, bytes: &[u8]) -> ValResult<'a, EitherTime<'a>> {
-    match Time::parse_bytes(bytes) {
+pub fn bytes_as_time<'a>(
+    input: &'a impl Input<'a>,
+    bytes: &[u8],
+    microseconds_overflow_behavior: MicrosecondsPrecisionOverflowBehavior,
```

Comment:
Any reason to go this way rather than `config: TimeConfig` argument?

Reply:
Just to be clear — we are viewing all rust code in pydantic-core as private, right? If that's the case, I'm fine with this either way. If this isn't published as a crate on cargo or whatever, that seems fine. If we did want to make it public, I'd be more inclined to use the `config: TimeConfig` approach, but I guess I prefer Adrian's approach if we can make breaking changes to this API with no consequences because I would agree it does do a better job of conveying precisely what configuration is meant to be controlled here.

Reply:
(Unless there are other options we could/should expose from `TimeConfig`, in which case, maybe that makes more sense. I haven't looked into it and don't intend to unless one of you suggests I should.)

Reply:
Correct, the python package is public (although I don't think it should be and I don't think anyone is depending on it) but the Rust code is 100% private